rebild project

OlegSsSs opened the pull request.

OlegSsSs committed:
rebild project

developer-aston commented on src/main/java/org/example/ArrayListOwn.java:
```diff
@@ -0,0 +1,215 @@
+     * Clears the list by remove all elements
+     */
+    public void clear() {
+//        Arrays.fill(arrayData, null);
```
закомментированный код

developer-aston commented on src/main/java/org/example/LinkedListOwn.java:
```diff
@@ -0,0 +1,269 @@
+    /**
+     *  Linked list Node. Node is a static nested class
+     */
+    static class Node {
```
```java
private static class Node<T>
```

developer-aston commented on src/main/java/org/example/LinkedListOwn.java:
```diff
@@ -0,0 +1,269 @@
+     */
+    static class Node {
+        int data;
+        Node next;
```
LinkedList это двусвязный список, ссылка есть на предыдущий и следующий элементы, поэтому нужно добавить  
```java
private Node<T> next;
```

OlegSsSs replied:
переделал реализацию на двусвязный список

developer-aston commented on src/main/java/org/example/LinkedListOwn.java:
```diff
@@ -0,0 +1,269 @@
+     * @param list to which is added data
+     * @param data the int data to be added
+     */
+    public static void add(LinkedListOwn list, int data) {
```
методы add, remove и т.п. должен принимать только добавляемое значение, но не list, у тебя принимает list и int. Необходимо в твоей реализации использовать обобщения - generics

OlegSsSs replied:
все исправил

developer-aston commented on src/main/java/org/example/LinkedListOwn.java:
```diff
@@ -0,0 +1,269 @@
+     * Sorts a linked list in ascending order using the insertion method
+     * @param list to which is need sorting
+     */
+    public static void sortList(LinkedListOwn list) {
```
здесь должен принимать Comparator

OlegSsSs replied:
исправил

developer-aston commented on src/main/java/org/example/LinkedListOwn.java:
```diff
@@ -0,0 +1,269 @@
+                currentIndex++;
+            }
+            if (currNode == null) {
+                System.out.println("is out of bounds for the list");
```
здесь должно быть выброшено исключение

OlegSsSs replied:
исправил

developer-aston commented on src/main/java/org/example/LinkedListOwn.java:
```diff
@@ -0,0 +1,269 @@
+     * Displays a list
+     * @param list will be displayed on screen
+     */
+    public static void printList(LinkedListOwn list) {
```
лишний метод, или toString для просмотра состава List или Iterator

OlegSsSs replied:
переопределил toString

developer-aston commented on src/main/java/org/example/LinkedListOwn.java:
```diff
@@ -0,0 +1,269 @@
+
+        if (currNode != null && currNode.data == value) {
+            list.head = currNode.next;
+            System.out.println(value + " value find and remove");
```
System.out.println не нужно использовать в таких методах. Для debug есть стандартные инструменты Intellij Idea

OlegSsSs replied:
убрал

developer-aston reviewed (commented):
папку target в .gitignore. Нужно переделать реализацию LinkedList с обобщения и учетом комментариев

OlegSsSs committed:
Update ArrayListOwn.java

был закомментированный код - исправил

OlegSsSs committed:
Update LinkedListOwn.java

все замечания устранил, переделал всю реализацию LinkedList с обобщениями и учетом комментариев

developer-aston reviewed (commented):
Функционал работает, всё хорошо поправил, также отлично что попрактиковался вручную написать сортировки!
